src/hooks/getAssetCapUsage.tsx

```diff
@@ -0,0 +1,128 @@
+};
+
+export type AssetCapUsageData = {
+  supplyCap: AssetCapHookData;
+  borrowCap: AssetCapHookData;
+  debtCeiling: AssetCapHookData;
+};
+
+type AssetLikeObject =
+  | ComputedReserveData
+  | (unknown & {
+      // Just whitelisting the fields used in this 'hook'
+      totalLiquidity: string;
+      supplyCap: string;
+      totalDebt: string;
+      borrowCap: string;
+      isolationModeTotalDebt: string;
+      debtCeiling: string;
+    })
+  | undefined;
+
+const getAssetCapUsage = (asset: AssetLikeObject): AssetCapUsageData => {
+  /*
+    Supply Cap Data
+    % of totalLiquidity / supplyCap
+    Set as zero if either is zero
+  */
+  let supplyCapUsage: number = asset
+    ? valueToBigNumber(asset.totalLiquidity).dividedBy(asset.supplyCap).toNumber() * 100
+    : 0;
+  supplyCapUsage = supplyCapUsage === Infinity ? 0 : supplyCapUsage;
+  const supplyCapReached = supplyCapUsage !== Infinity && supplyCapUsage >= 99.99;
+
+  /*
+    Borrow Cap Data
+    % of totalDebt / borrowCap
+    Set as zero if either is zero
+  */
+  let borrowCapUsage: number = asset
+    ? valueToBigNumber(asset.totalDebt).dividedBy(asset.borrowCap).toNumber() * 100
+    : 0;
+  borrowCapUsage = borrowCapUsage === Infinity ? 0 : borrowCapUsage;
+  const borrowCapReached = borrowCapUsage !== Infinity && borrowCapUsage >= 99.99;
+
+  /*
+    Debt Ceiling Data
+    % of isolationModeTotalDebt / debtCeiling
+    Set as zero if either is zero
+  */
+  let debtCeilingUsage: number = asset
+    ? valueToBigNumber(asset.isolationModeTotalDebt).dividedBy(asset.debtCeiling).toNumber() * 100
+    : 0;
+  debtCeilingUsage = debtCeilingUsage === Infinity ? 0 : debtCeilingUsage;
+  const debtCeilingReached = debtCeilingUsage >= 99.99;
+
+  /*
+    Aggregated Data
+  */
+  const assetCapUsageData: AssetCapUsageData = {
+    supplyCap: {
+      percentUsed: supplyCapUsage,
+      isMaxed: supplyCapReached,
+      // percentUsed: 99.9,
+      // isMaxed: false,
+      determineWarningDisplay: ({ supplyCap, icon }) =>
+        supplyCap ? <SupplyCapWarning supplyCap={supplyCap} icon={icon} /> : null,
+      determineTooltipDisplay: ({ supplyCap, useDefaultTooltip }) =>
+        supplyCap ? (
+          <SupplyCapTooltip supplyCap={supplyCap} useDefaultTooltip={useDefaultTooltip} />
+        ) : null,
+    },
+    borrowCap: {
+      percentUsed: borrowCapUsage,
+      isMaxed: borrowCapReached,
+      // percentUsed: 98.5,
+      // isMaxed: false,
+      determineWarningDisplay: ({ borrowCap, icon }) =>
+        borrowCap ? <BorrowCapWarning borrowCap={borrowCap} icon={icon} /> : null,
+      determineTooltipDisplay: ({ borrowCap, useDefaultTooltip }) =>
+        borrowCap ? (
+          <BorrowCapTooltip borrowCap={borrowCap} useDefaultTooltip={useDefaultTooltip} />
+        ) : null,
+    },
+    debtCeiling: {
+      percentUsed: debtCeilingUsage,
+      isMaxed: debtCeilingReached,
+      // percentUsed: 99.994,
+      // isMaxed: true,
```

Comment:
These commented out lines are what I've been using to quickly test scenarios for given screens/modals/warnings/etc - since not all live production networks are going to yield every possible scenario.